docs/learn/expert-guides/universal-profile/extend-profile-functionalities.md

```diff
@@ -0,0 +1,555 @@
+
+## Extending Functionalities
+
+### Setup
+
+Set up a new project with:
+
+```bash
+npm init
+```
+
+Then, install the dependencies
+
+```bash
+npm install ethers@v5 @lukso/lsp-smart-contracts@v0.14
+```
+
+### Step 1: Instantiate the Universal Profile
+
+#### Generate a Signer
+
+First, we need to generate a signer. This signer will be used to interact with the blockchain. We'll use a private key of the Universal Profile controller and an RPC URL.
+
+```js
+import { ethers } from 'ethers';
+
+// RPC URL (e.g., LUKSO testnet)
+const RPC_URL = 'https://rpc.testnet.lukso.network';
+
+// Replace with your private key
+const privateKey = 'your-private-key';
+
+const provider = new ethers.providers.JsonRpcProvider(RPC_URL);
+const signer = new ethers.Wallet(privateKey, provider);
+```
+
+#### Instantiate the Universal Profile Contract
+
+Now, we'll instantiate an instance of the Universal Profile contract.
+
+```js
+import { abi as UniversalProfileABI } from '@lukso/lsp-smart-contracts/artifacts/UniversalProfile.json';
+
+// Replace with the address of your Universal Profile
+const universalProfileAddress = 'your-universal-profile-address';
+
+const universalProfile = new ethers.Contract(
+  universalProfileAddress,
+  UniversalProfileABI,
+  signer,
+);
+```
+
+### Step 2: Create the Extension Contract
+
+Next, we will create a Solidity contract named `TipMe` which will inherit from [`LSP17Extension`](../../../contracts/contracts/LSP17ContractExtension/LSP17Extension.md) and have a function `tipMe(..)` that emits an event `Tipped`.
+
+```solidity
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@lukso/lsp-smart-contracts/contracts/LSP17ContractExtension/LSP17Extension.sol";
+
+contract TipMe is LSP17Extension {
+    event Tipped(address tipper);
+
+    function tipMe() public payable {
+      // _extendableMsgSender() is a method inherited from LSP17Extension that
+      // extract the address who initially called the Universal Profile with the tipMe selector
+      // `msg.sender` in this case will be the Universal Profile address/
+      emit Tipped(_extendableMsgSender());
+    }
+
+    // You can access the value sent to the UniversalProfile with
+    // _extendableMsgValue() if the value is being preserved in the UniversalProfile
+    // or with msg.value in case the UniversalProfile is forwarding the value to the extension
+
+}
+```
+
+The contract can be compiled with any blockchain development environment like Hardhat or Foundry and extract the abi and bytecode. Create `TipMe.json` in the same directory of the main file.
```

Comment:
This is where I think it would be nice to introduce the getting started guide earlier, and put the guide in that perspective. as the getting started already has the script, the luksoTestnet env in hardat networks etc.